Shared/mods/deathmatch/logic/luadefs/CLuaCryptDefs.cpp

```diff
@@ -516,6 +667,73 @@ int CLuaCryptDefs::DecodeString(lua_State* luaVM)
                 }
                 return 1;
             }
+            case StringEncodeFunction::RSA:
+            {
+                SString& key = options["key"];
+
+                if (key.empty())
+                {
+                    m_pScriptDebugging->LogCustom(luaVM, "Invalid value for field 'key'");
+                    lua::Push(luaVM, false);
+                    return 1;
+                }
+
+                // Async
+                if (VERIFY_FUNCTION(luaFunctionRef))
+                {
+                    CLuaMain* pLuaMain = m_pLuaManager->GetVirtualMachine(luaVM);
+                    if (pLuaMain)
+                    {
+                        CLuaShared::GetAsyncTaskScheduler()->PushTask<std::pair<SString, bool>>(
+                            [data, key]
+                            {
+                                try
+                                {
+                                    return std::make_pair(SharedUtil::RsaDecode(data, key), true);
+                                }
+                                catch (const CryptoPP::Exception& ex)
+                                {
+                                    return std::make_pair(SString(ex.GetWhat()), false);
+                                }
```

Comment:
This is better exception handling than just ignoring it and returning false. However, I'm not sure if we want to expose scripts to the implementation-defined messages of CryptoPP. According to the [docs](https://www.cryptopp.com/docs/ref/class_exception.html#ac2c2e8a3c1871c532660252cd1894b90), it should be feasible to get an error type and control what string is logged ourselves.

Reply:
That was @Pirulax idea to expose exception cause

Reply:
I agree with exposing the exception cause somehow. But I'm not sure if we want to potentially make backwards compatibility more complicated just because some script decided to rely on some precise warning text. This is a pretty minor and far-fetched concern, though, and I'd be happy with this merged as-is.